recipes/default.rb

```diff
@@ -15,14 +15,14 @@
   node['rackspace_users']['data_bag'],
   node['rackspace_users']['data_bag_item']).to_hash.select { |user, user_data| user != 'id' }
 
-node_tags = node['rackspace_users']['node_tags']
+node_groups = node['rackspace_users']['node_groups']
 
 groups = {}
 
 users.each do |username, user_data|
-  ## The action (if defined in the data bag) should be overwritten to 'remove' if the user's node tags and node's tags don't intersect.
-  ## If the user doesn't declare any tags then it is assumed that is present on all nodes.
-  user_data['action'] = 'remove' if user_data['node_tags'] && (user_data['node_tags'] & node_tags).empty?
+  ## The action (if defined in the data bag) should be overwritten to 'remove' if not all the user's node_groups are in node's node_groups.
+  ## If the user doesn't declare node_groups then it is assumed that is present on all nodes.
+  user_data['action'] = 'remove' if user_data['node_groups'] && !(user_data['node_groups'] - node_groups).empty?
```

Comment:
The problem with a toggle is that you then can't use both.  Suppose a node has groups `dev`, `web`, and `drupal`.  User `joedev` should have access to dev servers running drupal, so you toggle the node to 'all must match' mode and set `joedev`'s groups to `dev` and `drupal`.  Now suppose user `appadmin` should have access to all servers running web apps, so it has groups `drupal`, `wordpress`, and `moodle`.  This requires the node be in the 'any must match' mode though, since it doesn't have groups for wordpress or moodle and the `appadmin` user won't get created.

If you toggle based on the user instead of the node, then in 'any' mode, `joedev` suddenly gets access to all the production drupal servers as well since his `drupal` group matched.  In 'all' mode, what if `joedev` also has a `db` group that he needs to access the database servers?  Now he can't get into the drupal servers since they don't have a `db` group.

You could potentially mitigate this situation in the databag by having "node group groups", like this:
```
"users": {
    "joedev": {
        "node_groups": [
            ['dev', 'drupal'],
            ['db']
        ]   
    }   
}
```
The `joedev` user now gets created on any server that has both the `dev` and `drupal` groups, or that has the `db` group.  More generally, all groups in a list must match, but only one of the lists has to match.

With such complex logic though, it might make more sense to just use 'any matching group' and leave the rest of the logic to recipes.